os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -0,0 +1,791 @@
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_get_hmac(hal_hmac_type mode, hal_data *input, uint32_t key_idx, hal_data *hmac)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = input;
+	in_output.output_data = hmac;
+
+	ret = ameba_hal_get_hmac(mode, &in_output, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_rsa_sign_md(hal_rsa_mode mode, hal_data *hash, uint32_t key_idx, hal_data *sign)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = hash;
+	in_output.output_data = sign;
+
+	ret = ameba_hal_rsa_sign_md(mode, &in_output, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_rsa_verify_md(hal_rsa_mode mode, hal_data *hash, hal_data *sign, uint32_t key_idx)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = hash;
+	in_output.output_data = sign;
+
+	ret = ameba_hal_rsa_verify_md(mode, &in_output, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_ecdsa_sign_md(hal_ecdsa_mode mode, hal_data *hash, uint32_t key_idx, hal_data *sign)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = hash;
+	in_output.output_data = sign;
+
+	ret = ameba_hal_ecdsa_sign_md(mode, &in_output, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_ecdsa_verify_md(hal_ecdsa_mode mode, hal_data *hash, hal_data *sign, uint32_t key_idx)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = hash;
+	in_output.output_data = sign;
+
+	ret = ameba_hal_ecdsa_verify_md(mode, &in_output, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_dh_generate_param(uint32_t dh_idx, hal_dh_data *dh_param)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	dh_idx = rtl_index_convt(dh_idx);
+	if (dh_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	if (!dh_param->pubkey->data) {
```

Comment:
`dh_param` and `dh_param->pubkey` can be dereferencing. Please check these before using.

Reply:
Sorry, it is a wrong reference from SSS. I will remove it.